src/Common/src/Interop/Unix/System.Net.Security.Native/Interop.Gss.cs

```diff
@@ -0,0 +1,336 @@
+            SafeGssNameHandle desiredName,
+            bool isInitiate,
+            out SafeGssCredHandle outputCredHandle);
+
+        [DllImport(Interop.Libraries.SecurityNative, CharSet = CharSet.Ansi)]
+        internal static extern Status GssAcquireCredWithPassword(
+            out Status minorStatus,
+            SafeGssNameHandle desiredName,
+            string password,
+            bool isInitiate,
+            out SafeGssCredHandle outputCredHandle);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssReleaseCred(
+            out Status minorStatus,
+            ref IntPtr credHandle);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssInitSecContext(
+            out Status minorStatus,
+            SafeGssCredHandle initiatorCredHandle,
+            ref SafeGssContextHandle contextHandle,
+            bool isNtlm,
+            SafeGssNameHandle targetName,
+            uint reqFlags,
+            byte[] inputBytes,
+            int inputLength,
+            SafeGssBufferHandle outputToken,
+            out uint retFlags);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssAcceptSecContext(
+            out Status minorStatus,
+            ref SafeGssContextHandle contextHandle,
+            SafeGssCredHandle initiatorCredHandle,
+            SafeGssBufferHandle inputToken,
+            SafeGssBufferHandle outputToken,
+            out uint retFlags);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssDeleteSecContext(
+            out Status minorStatus,
+            ref IntPtr contextHandle);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssWrap(
+            out Status minorStatus,
+            SafeGssContextHandle contextHandle,
+            bool isEncrypt,
+            byte[] inputBytes,
+            int offset,
+            int count,
+            SafeGssBufferHandle outputMessageBuffer);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssUnwrap(
+            out Status minorStatus,
+            SafeGssContextHandle contextHandle,
+            byte[] inputBytes,
+            int offset,
+            int count,
+            SafeGssBufferHandle outputMessageBuffer);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssInquireSourceName(
+            out Status minorStatus,
+            SafeGssContextHandle contextHandle,
+            out SafeGssNameHandle srcName);
+
+        [DllImport(Interop.Libraries.SecurityNative)]
+        internal static extern Status GssDisplayName(
+            out Status minorStatus,
+            SafeGssNameHandle inputName,
+            SafeGssBufferHandle outputNameBuffer);
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct gss_buffer_desc
```

Comment:
I think we should be able to get rid of this dependency with a strategy like foll.
1. Wherever gss_buffer_desc is getting set in shim layer, also return the length
2. Wherever gss_buffer_desc.Value and Length are getting used by callers, we can create SafeGssBufferHandle.Copy methods which take the length from Step 1 as input, do an allocation in managed layer, call the shim to do a memcpy to the allocated buffer from gss_buffer_desc